lib/deep-required/index.ts

```diff
@@ -18,6 +20,12 @@ export type DeepRequired<Type> = Type extends Error
   ? WeakSet<DeepRequired<Values>>
   : Type extends Promise<infer Value>
   ? Promise<DeepRequired<Value>>
+  : Type extends AnyArray<infer Values>
+  ? Type extends Tuple
+    ? { [Key in keyof Type]-?: DeepRequired<Type[Key]> }
+    : Type extends Array<Values>
```

Comment:
It seems a real shame that we have to explicitly deal with both Array and ReadonlyArray here, but I'm not quite sure how to handle both types more implicitly.

Reply:
It's possible to unite it with another type, but for simplicity it's left as is so it will be clear what happens for new people

Sometimes we have to do different things for readonly, normal arrays and tuples, that's why we still do it manually

If you have any suggestion how to generally handle it better, I'm more than open to address it in another issue

Reply:
I definitely don't - was kinda hoping I was missing something 😁.